DevCycle.SDK.Server.Common/API/DevCycleProvider.cs

```diff
@@ -0,0 +1,78 @@
+    {
+        private DevCycleBaseClient Client { get; }
+
+        public DevCycleProvider(DevCycleBaseClient client)
+        {
+            Client = client;
+        }
+
+        public override Metadata GetMetadata()
+        {
+            return new Metadata(Client.SdkPlatform);
+        }
+
+        private async Task<ResolutionDetails<T>> EvaluateDevCycle<T>(string flagKey, T defaultValue,
+            EvaluationContext context = null)
+        {
+            var user = DevCycleUser.FromEvaluationContext(context);
+            var variable = await Client.Variable(user, flagKey, defaultValue);
+            return variable.GetResolutionDetails();
+        }
+
+        public override async Task<ResolutionDetails<bool>> ResolveBooleanValue(string flagKey, bool defaultValue,
+            EvaluationContext context = null)
+        {
+            return await EvaluateDevCycle(flagKey, defaultValue, context);
+        }
+
+        public override async Task<ResolutionDetails<string>> ResolveStringValue(string flagKey, string defaultValue,
+            EvaluationContext context = null)
+        {
+            return await EvaluateDevCycle(flagKey, defaultValue, context);
+        }
+
+        public override async Task<ResolutionDetails<int>> ResolveIntegerValue(string flagKey, int defaultValue,
+            EvaluationContext context = null)
+        {
+            return await EvaluateDevCycle(flagKey, defaultValue, context);
+        }
+
+        public override async Task<ResolutionDetails<double>> ResolveDoubleValue(string flagKey, double defaultValue,
+            EvaluationContext context = null)
+        {
+            return await EvaluateDevCycle(flagKey, defaultValue, context);
+        }
+
+        public override async Task<ResolutionDetails<Value>> ResolveStructureValue(string flagKey, Value defaultValue,
+            EvaluationContext context = null)
+        {
+            if (!defaultValue.IsStructure)
+                throw new System.Exception("Cannot call ResolveStructureValue with non-structure Value's");
+            var jsonString = JsonSerializer.Serialize(defaultValue,
+                new JsonSerializerOptions() { Converters = { new OpenFeatureValueJsonConverter() } });
+
+            var newtonsoftJObj = Newtonsoft.Json.JsonConvert.DeserializeObject(jsonString);
+            var user = DevCycleUser.FromEvaluationContext(context);
+            var variable = await Client.Variable(user, flagKey, (JObject)newtonsoftJObj);
+            var jsonValue = Newtonsoft.Json.JsonConvert.SerializeObject(variable.Value);
+            var openFeatureValue = JsonSerializer.Deserialize<Value>(jsonValue,
+                new JsonSerializerOptions() { Converters = { new OpenFeatureValueJsonConverter() } });
+            
+            return new ResolutionDetails<Value>(flagKey, openFeatureValue, ErrorType.None,
+                variable.IsDefaulted ? Reason.Default : Reason.TargetingMatch);
```

Comment:
Unless I'm misunderstanding you could remove this check by calling `.GetResolutionDetails()` on the new object. Might be worth it since `variable.IsDefaulted ? Reason.Default : Reason.TargetingMatch)` seems like a domain logic duplication.

Reply:
The reason for that not working is the `Variable<T>` generic type param doesn't match the types - as it'd expect to then create a `Variable<Value>` which isn't a valid datatype to the native SDK. Hence having to do this manual reconstruction for JSON since we use Newtonsoft as the root JSON lib - and the equivalent would be `Variable<JObject>`